### DISCUSSION TITLE
Use IdentityUser in CleanAchitecture

### DISCUSSION THREAD

**Discussion by alnebras** [jasontaylordev/CleanArchitecture#682] [Q&A]
Hello,

I would like a user class that inherits from IdentityUser,

the problem is user class is found in the infrastructure (to implement identity in the context) layer so I can't implement login and registration on the application layer because it's not inherited from the infrastructure.

**Comment by neozhu** [comment]
You can refer to this https://docs.microsoft.com/en-us/aspnet/core/security/authentication/customize-identity-model?view=aspnetcore-6.0
and you can use these two classes SignInManager<ApplicationUser>,UserManager<ApplicationUser> to implement the registration and login functions you want 
like as my project code: https://github.com/neozhu/CleanArchitectureWithBlazorServer/blob/main/src/Blazor.Server.UI/Pages/Authentication/Login.razor

Hope this helps you
